pennylane/templates/subroutines/select.py

```diff
@@ -1086,3 +1086,91 @@ def _select_decomp_unary(*_, ops, control, work_wires, partial, **__):
 
 
 add_decomps(Select, _select_decomp_unary)
+
+
+# Decomposition of Select using one work wire to control the target operations
+
+
+def _select_resources_multi_control_work_wire(op_reps, num_control_wires, num_work_wires, partial):
+    resources = defaultdict(int)
+
+    if partial:
+        if len(op_reps) == 1:
+            resources[_multi_controlled_rep(op_reps[0], 1, [1], num_work_wires - 1)] += 1
+            resources[
+                _multi_controlled_rep(
+                    resource_rep(X), num_control_wires, [0] * num_control_wires, num_work_wires - 1
+                )
+            ] += 2
+        else:
+            # Use dummy control values, we will only care about the length of the outputs
+            ctrls_and_ctrl_states = _partial_select(len(op_reps), list(range(num_control_wires)))
+            for (ctrl_, ctrl_state), rep in zip(ctrls_and_ctrl_states, op_reps):
+                resources[_multi_controlled_rep(rep, 1, [1], num_work_wires - 1)] += 1
+                resources[
+                    _multi_controlled_rep(
+                        resource_rep(X), len(ctrl_), ctrl_state, num_work_wires - 1
+                    )
+                ] += 2
+    else:
+        state_iterator = product([0, 1], repeat=num_control_wires)
+
+        for state, rep in zip(state_iterator, op_reps):
+
+            resources[_multi_controlled_rep(rep, 1, [1], num_work_wires - 1)] += 1
+            resources[
+                _multi_controlled_rep(resource_rep(X), num_control_wires, state, num_work_wires - 1)
+            ] += 2
+    return dict(resources)
+
+
+# pylint: disable=unused-argument
+def _work_wire_condition(op_reps, num_control_wires, partial, num_work_wires):
+    return num_work_wires >= 1
```

Comment:
Not sure I understand the condition. I thought the `work_wires` in `Select` were used to decompose results `MultiControlledX`. Does this condition mean that when a user inputs `work_wires` they'll get used for this decomposition instead of being used for `MultiControlledX`?

Reply:
Each decomposition will chose how to manage the work_wires
In the naive one, all wires are used in the MultiControlX
In the new one, all wires (-1) are used in the MulticontrolX, the extra one is used with another purpose
In the unary decomposition, they follow a different structure